packages/webapp/src/containers/Animals/AddAnimals/useAnimalOptions.ts

```diff
@@ -106,14 +108,32 @@ export const useAnimalOptions = (...optionTypes: OptionType[]) => {
   }
 
   if (optionTypes.includes('use')) {
-    options.useOptions = uses.map((animalType) => ({
-      default_type_id: animalType.default_type_id,
-      uses: animalType.uses.map((use) => ({
-        value: use.id,
-        label: t(`animal:USE.${use.key}`),
-        key: use.key,
-      })),
-    }));
+    options.useOptions = defaultTypes.map((animalType) => {
+      return {
+        default_type_id: animalType.id,
+        uses: typeUseRelationships
+          .filter((typeUse) => typeUse.default_type_id === animalType.id)
```

Comment:
Essentially we remove all of the controller logic and only have to add in this filter, and the all uses option below.